utils/symbol.c

```diff
@@ -1298,13 +1298,12 @@ static void save_module_symbol(struct symtab *stab, const char *symfile,
 		fprintf(fp, "%016"PRIx64" %c %s\n", sym->addr - offset,
 			(char) sym->type, sym->name);
 	}
-	if (i > 0) {
-		struct sym *prev = &stab->sym[i - 1];
+	
+	struct sym *prev = &stab->sym[i - 1];
```

Comment:
While at it, I think it's better renaming this to 'last'.

Reply:
Like this way?

```
struct sym *last = &stab->sym[i - 1];

fprintf(fp, "%016"PRIx64" %c __%ssym_end\n",
	last->addr + last->size - offset,
	(char) last->type, last_was_plt ? "dyn" : "");
```

Reply:
Right but excluding the last one - 'prev_was_plt' which is updated in the loop.

Reply:
Aha, Like this way?
```
struct sym *last = &stab->sym[i - 1];

fprintf(fp, "%016"PRIx64" %c __%ssym_end\n",
	last->addr + last->size - offset,
	(char) last->type, prev_was_plt ? "dyn" : "");
```

Reply:
Yep.  While it's effectively same, I slightly prefer this:

```c
struct sym *last = &stab->sym[stab->nr_sym - 1];

...
```

Reply:
Okay.
I'll fix it.